src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchProfile.cs

```diff
@@ -84,13 +85,71 @@ public LaunchProfile(ILaunchProfile existingProfile)
                 _otherSettings = value;
             }
         }
-
+               
         /// <summary>
         /// Compares two profile names. Using this function ensures case comparison consistency
         /// </summary>
         public static bool IsSameProfileName(string name1, string name2)
         {
             return string.Equals(name1, name2, StringComparison.Ordinal);
         }
+
+        /// <summary>
+        /// Compares two IDebugProfiles to see if they contain the same values. Note that it doesn't compare
+        /// the kind field unless includeProfileKind is true
+        /// </summary>
+        public static bool ProfilesAreEqual(ILaunchProfile debugProfile1, ILaunchProfile debugProfile2, bool includeProfileKind)
+        {
+            // Same instance better return they are equal
+            if (debugProfile1 == debugProfile2)
+            {
+                return true;
+            }
+
+            if (!string.Equals(debugProfile1.Name, debugProfile2.Name, StringComparison.Ordinal) ||
+               !string.Equals(debugProfile1.CommandName, debugProfile2.CommandName, StringComparison.Ordinal) ||
+               !string.Equals(debugProfile1.ExecutablePath, debugProfile2.ExecutablePath, StringComparison.Ordinal) ||
+               !string.Equals(debugProfile1.CommandLineArgs, debugProfile2.CommandLineArgs, StringComparison.Ordinal) ||
+               !string.Equals(debugProfile1.WorkingDirectory, debugProfile2.WorkingDirectory, StringComparison.Ordinal) ||
+               !string.Equals(debugProfile1.LaunchUrl, debugProfile2.LaunchUrl, StringComparison.Ordinal) ||
+               (debugProfile1.LaunchBrowser != debugProfile2.LaunchBrowser) ||
+               (includeProfileKind && debugProfile1.CommandName != debugProfile2.CommandName))
```

Comment:
To be complete, this needs to also validate OtherSettings. Since we don't have any editing of other settings (and mostly likely won't in the future), you will need to preserve them when edits are made in the UI (Kind of like how you saved GlobalSettings)